tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPSquareMirroringTest.java

```diff
@@ -0,0 +1,278 @@
+         server.getConfiguration().addAMQPConnection(amqpConnection);
+      }
+
+      {
+         AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "2to1", "tcp://localhost:" + AMQP_PORT).setRetryInterval(100);
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement().setCanForwardMessages(false));
+         server_2.getConfiguration().addAMQPConnection(amqpConnection);
+         amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "2to3", "tcp://localhost:" + AMQP_PORT_3).setRetryInterval(100);
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement());
+         server_2.getConfiguration().addAMQPConnection(amqpConnection);
+      }
+
+      {
+         AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "3to2", "tcp://localhost:" + AMQP_PORT_2).setRetryInterval(100);
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement());
+         server_3.getConfiguration().addAMQPConnection(amqpConnection);
+         amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "3to4", "tcp://localhost:" + AMQP_PORT_4).setRetryInterval(100);
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement().setCanForwardMessages(false));
+         server_3.getConfiguration().addAMQPConnection(amqpConnection);
+      }
+
+      {
+         AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "4to1", "tcp://localhost:" + AMQP_PORT).setRetryInterval(100);
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement());
+         server_4.getConfiguration().addAMQPConnection(amqpConnection);
+         amqpConnection = new AMQPBrokerConnectConfiguration(getTestMethodName() + "4to3", "tcp://localhost:" + AMQP_PORT_3).setRetryInterval(100);
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement().setCanForwardMessages(false));
+         server_4.getConfiguration().addAMQPConnection(amqpConnection);
+      }
+
+      server.start();
+      server_2.start();
+      server_3.start();
+      server_4.start();
+
+      createAddressAndQueues(server);
+      Wait.assertTrue(() -> server.locateQueue(getQueueName()) != null);
+      Wait.assertTrue(() -> server_2.locateQueue(getQueueName()) != null);
+      Wait.assertTrue(() -> server_3.locateQueue(getQueueName()) != null);
+      Wait.assertTrue(() -> server_4.locateQueue(getQueueName()) != null);
+
+      Queue q1 = server.locateQueue(getQueueName());
+      assertNotNull(q1);
+
+      Queue q2 = server.locateQueue(getQueueName());
+      assertNotNull(q2);
+
+      Queue q3 = server.locateQueue(getQueueName());
+      assertNotNull(q3);
+
+      Queue q4 = server.locateQueue(getQueueName());
+      assertNotNull(q4);
+
+      ConnectionFactory factory = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:" + AMQP_PORT);
```

Comment:
can you make sure this test works with other protocols?


you can either run all 3 protocols (AMQP, CORE, OPENWIRE) with a parameter, or make the selection random if this test is a long running one (I added recently an method on ActiveMQTestBase called randomProtocol())

Reply:
I tried to use a `randomProtocol()`:
![image](https://github.com/user-attachments/assets/e62e6394-0055-44f2-9bb3-64d4e6101e53)

and then this error happened: 
![image](https://github.com/user-attachments/assets/6d269d66-18f4-48ed-ada5-d19b7ced6052)

Any idea on what could be going on?

Reply:
This got fixed, I'm now using the randomProtocols